### PR TITLE
COMP: Disable CMake user registry lookup by find_package.

### PR DESCRIPTION
**Pull request by jcfr**
See #1469

## PR Checklist

- :no_entry_sign: [Makes breaking changes](https://github.com/InsightSoftwareConsortium/ITK/blob/master/CONTRIBUTING.md#breaking-changes)
- :no_entry_sign: [Makes design changes](https://github.com/InsightSoftwareConsortium/ITK/blob/master/CONTRIBUTING.md#design-changes)
- :no_entry_sign: Adds the License notice to new files.
- :no_entry_sign: Adds Python wrapping.
- :no_entry_sign: Adds tests and baseline comparison (quantitative).
- :no_entry_sign: [Adds test data](https://github.com/InsightSoftwareConsortium/ITK/blob/master/Documentation/UploadBinaryData.md).
- :no_entry_sign: Adds Examples to [ITKExamples](https://github.com/InsightSoftwareConsortium/ITKExamples)
- :no_entry_sign: Adds Documentation.

Refer to the [ITK Software Guide](https://itk.org/ItkSoftwareGuide.pdf) for
further development details if necessary.

### CODE REVIEW

**Comment by thewtex** [comment]
@jcfr thanks for taking the time to file the issue and submit patch on this tricky time sink :potable_water: that others will hopefully not need to encounter :pray: :1st_place_medal: 

Per the link in the report, since `CMAKE_FIND_NO_PACKAGE_REGISTRY ` [is deprecated](https://cmake.org/cmake/help/latest/variable/CMAKE_FIND_PACKAGE_NO_PACKAGE_REGISTRY.html#variable:CMAKE_FIND_PACKAGE_NO_PACKAGE_REGISTRY), should [`CMAKE_FIND_USE_PACKAGE_REGISTRY`](https://cmake.org/cmake/help/latest/variable/CMAKE_FIND_USE_PACKAGE_REGISTRY.html#variable:CMAKE_FIND_USE_PACKAGE_REGISTRY`]) be used instead?

**Comment by jcfr** [comment]
> should CMAKE_FIND_USE_PACKAGE_REGISTRY be used instead?

Until ITK depends on CMake >= 3.16, I suggest we keep using the deprecated variable. Indeed, the new one `CMAKE_FIND_USE_PACKAGE_REGISTRY` was introduced in CMake 3.16
See https://cmake.org/cmake/help/latest/release/3.16.html?highlight=cmake_find_use_package_registry#commands

We could check for the version of CMake and set one or the other ?

**Comment by hjmjohnson** [comment]
> We could check for the version of CMake and set one or the other ?

I think we should check version numbers.  That way when we require CMake 3.16 (in a few years :) ), it will be easy to remove the outdated variable name.

**Comment by jcfr** [comment]
> I think we should check version numbers. That way when we require CMake 3.16 (in a few years :) ), it will be easy to remove the outdated variable name.

@hjmjohnson  Great idea :bulb: , let me know what you think.

File: CMakeLists.txt [hunk 1]
```diff
@@ -242,8 +242,15 @@ set(CMAKE_DEBUG_POSTFIX "" CACHE STRING "Generate debug library name with a post
...
 
 # Configure find_package behavior
-if(NOT DEFINED CMAKE_FIND_PACKAGE_NO_PACKAGE_REGISTRY)
-  set(CMAKE_FIND_PACKAGE_NO_PACKAGE_REGISTRY 1)
+if("${CMAKE_VERSION}" VERSION_GREATER_EQUAL "3.16")
+  if(NOT DEFINED CMAKE_FIND_USE_PACKAGE_REGISTRY)
+    set(CMAKE_FIND_USE_PACKAGE_REGISTRY 1)
+  endif()
+else()
+  message("HERE")
```
Comment (hjmjohnson) [inline CMakeLists.txt]: OOPS:  Remove "HERE" message.

Comment (jcfr) [reply CMakeLists.txt]: Good catch. Thanks

File: CMakeLists.txt [hunk 2]
```diff
@@ -242,8 +242,14 @@ set(CMAKE_DEBUG_POSTFIX "" CACHE STRING "Generate debug library name with a post
 mark_as_advanced(CMAKE_DEBUG_POSTFIX)
 
 # Configure find_package behavior
-if(NOT DEFINED CMAKE_FIND_PACKAGE_NO_PACKAGE_REGISTRY)
-  set(CMAKE_FIND_PACKAGE_NO_PACKAGE_REGISTRY 1)
+if("${CMAKE_VERSION}" VERSION_GREATER_EQUAL "3.16")
+  if(NOT DEFINED CMAKE_FIND_USE_PACKAGE_REGISTRY)
+    set(CMAKE_FIND_USE_PACKAGE_REGISTRY 1)
```
Comment (dzenanz) [inline CMakeLists.txt]: I thought the logic here is positive, so this should be `FALSE`, not `1`.

Comment (jcfr) [reply CMakeLists.txt]: An other good catch ... :pray:  Thanks for the review

**Comment by jcfr** [comment]
Thanks for the review. I will integrate once the CI finishes.

**Pull request merged**